diff --git a/drivers/gpu/drm/canaan/canaan_vo.c b/drivers/gpu/drm/canaan/canaan_vo.c
--- a/drivers/gpu/drm/canaan/canaan_vo.c
+++ b/drivers/gpu/drm/canaan/canaan_vo.c
@@ -220,5 +220,6 @@ static void canaan_vo_update_osd(struct canaan_vo *vo,
 	switch (fb->format->format) {
+	case DRM_FORMAT_XRGB8888:
 	case DRM_FORMAT_ARGB8888:
 		reg_val = 0x53;
 		rb_swap = 0x4F;
 		break;
@@ -716,7 +717,7 @@ static const uint32_t video_plane_formats[] = {
 };
 
 static const uint32_t osd_plane_formats[] = {
-	DRM_FORMAT_ARGB8888, DRM_FORMAT_ARGB4444, DRM_FORMAT_ARGB1555,
+	DRM_FORMAT_XRGB8888, DRM_FORMAT_ARGB8888, DRM_FORMAT_ARGB4444, DRM_FORMAT_ARGB1555,
 	DRM_FORMAT_RGB888,   DRM_FORMAT_RGB565, DRM_FORMAT_BGR888
 };
 
